src/components/MailberryForm.tsx

```diff
@@ -1,155 +1,192 @@
-          headers: { 'Content-Type': 'application/json' },
-          body: JSON.stringify(formData)
-        });
-        if (response.ok) {
-          setShowThanksMessage(true);
-        } else {
-          setShowErrorMessage(true);
-        }
-      } catch (error) {
-        setShowErrorMessage(true);
-      }
-    }
-
-    setIsSubmitting(false);
-  };
-
-  return (
-    <div className={`mailberry-form mailberry-${format}`} style={{ display: showForm ? 'block' : 'none' }}>
-      <form onSubmit={handleSubmit}>
-        {
-        fields.map((field, index) => (
-          <div key={index} className="MBinput-wrapper">
-            <label className="MBlabel">
-              {field.label}{field.required && '*'}
-            </label>
-            <input
-              type={field.type}
-              name={field.label}
-              required={field.required}
-              className="MBinput"
-              value={formData[field.label.toLowerCase()] || ''}
-              onChange={(e) => handleInputChange(field.label, e.target.value)}
-            />
-          </div>
-        ))
-      }
-      {/* Submit Button */}
-      <div className="MBbtn-wrapper">
-        <button type="submit" className="MBbtn" disabled={isSubmitting}>
-          {text.button}
-        </button>
-      </div>
-
-      {/* Optional: Loader while submitting */}
-      {isSubmitting && (
-        <div className="MBspinner-wrapper">
-          <div className="MBspinner"></div>
-          {/* You can add a spinner icon or animation here */}
-        </div>
-      )}
-
-      {/* Optional: Success and Error Messages */}
-      {showThanksMessage && (
-        <div className="MBthank-you-wrapper">
-          <p className="MBthank-you-message">{text.thanksMessage}</p>
-        </div>
-      )}
-      {showErrorMessage && (
-        <div className="MBerror-wrapper">
-          <p className="MBerror-message">
-            Something went wrong.
-          </p>
-        </div>
-      )}
-      </form>
-    </div>
-  );
-};
-
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { FieldType, FormPopupOptions, FormatOptions, MailberryFormCSSProps, formFormat } from '../types';
+import { css } from '../utils/css-generator';
+import MailberryFormFieldComponents from "./MailberryFormField";
+import MailberryFormPopup from './MailberryPopupOption';
+import MailberryFormSnippet from './MailberrySnippetOption';
+
+type ContextProps = {
+  fields: FieldType[];
+  setFields: React.Dispatch<React.SetStateAction<FieldType[]>>;
+  emptyFields: boolean;
+  invalidEmail: boolean;
+  isSubmitting: boolean;
+  setIsSubmitting: React.Dispatch<React.SetStateAction<boolean>>;
+  isSubmitted: boolean;
+  setIsSubmitted: React.Dispatch<React.SetStateAction<boolean>>;
+  showErrorMessage: boolean;
+  showThanksMessage: boolean;
+}
+
+export const FormContext = createContext({} as ContextProps);
+
+const API_FORM_URL = 'https://backend.mailberry.ai/public'

```

Comment:
how should this be worked out in staging and dev?

Reply:
Do you think it is a good idea to introduce a function that reads the environment variable and, depending on the value, returns the corresponding url?

Reply:
yes, it has to be on run time, inside a function.